cmd/thanos/downsample.go

```diff
@@ -55,15 +57,23 @@ func runDownsample(
 	httpAddr string,
 	dataDir string,
 	gcsBucket string,
+	s3Config *s3.Config,
 	syncDelay time.Duration,
+	component string,
 ) error {
-	gcsClient, err := storage.NewClient(context.Background())
-	if err != nil {
-		return errors.Wrap(err, "create GCS client")
+	var (
+		bkt objstore.Bucket
+		err error = nil
+		// closeFn gets called when the sync loop ends to close clients, clean up, etc
+		closeFn = func() error { return nil }
+	)
+
+	if gcsBucket != "" || s3Config.Validate() == nil {
```

Comment:
Don't get this, why it needs to be here and in every command?

Basically, we already check it in `client.NewBucket` and return `ErrNotFound` if that happens.. so we can in every command handle NotFound bucket configuration properly. 

Some commands need to error immediately if ErrNotFound happens (their require bucket to work):
- bucket
- downsample
- compactor
- store

Sidecar as the only one can be OK if ErrNotFound. In that case we just not create shipper.

Also by doing it, we will be able to delete these vars which can be confusing (and trigger potential null pointer panics):
```
	var (
		bkt objstore.Bucket
		err error = nil

		closeFn = func() error { return nil }
	)
```